### ISSUE TITLE
Final model large size

### ISSUE THREAD

**Issue by kabirmdasraful** [awslabs/autogluon#943]
Hi,
First of all thanks to all developers for providing such a great package.  I am facing a space problem when `presets='best_quality'` is selected. The reason what I understand is that it tries different types of model and save all of them. 
Is there any way to save only the best model in one single file as H2O-3 package does? Also saving option not only in `.pkl` but also in `binary` form?

**Comment by Innixma** [comment]
If disk space is a concern, consider using:

`presets=['high_quality_fast_inference_only_refit', 'optimize_for_deployment']`.
This should be ~20x-40x+ reduction in disk space used.

If that is still too much, use:
`presets=['good_quality_faster_inference_only_refit', 'optimize_for_deployment']`.
This will be ~100x+ reduction in disk space used.

If you'd like to use only a particular model and delete all unnecessary models, call:
`predictor.delete_models(models_to_keep='best', dry_run=False)`

After this is done, you can consider `predictor` as a single model. Pickle is already binary form, so I'm unsure what more could be done besides the saving of individual models to their own custom binary form, but I doubt this would save all that much space.

We are also working on methods to compress files in AG for generic space saving at the cost of inference speed when loading from disk: https://github.com/awslabs/autogluon/issues/581

**Comment by kabirmdasraful** [comment]
@Innixma Thanks for the clarification.  I have another question. I am not sure if need to open a separate issue for this.

`predictor = task.fit(train_data, label='class',time_limits = 2*60)` if I run this code it automaticlly starts saving the model which sometimes inconvenience for user. Mostly as a scikit-learn user, we are used to saving the model only after if the user thinks the model is perfect using a command like `pickle.dump(predictor, open('model.pkl', 'wb'))`.

Is there any way to stop Autosaving the model before the user wants to save it? 

Or In the future is there any plan to do so?

Sorry for asking so many questions. As the package is new there are so many things still unclear.

**Comment by Innixma** [comment]
We currently don't support delaying model saving, as otherwise we would need to persist all trained models in memory at once, which can lead to numerous memory error related problems if the user isn't careful. This may be added in a future release, but isn't on our current roadmap. My suggestion is to increase disk space or avoid using models that take up a large amount of space (i.e. KNN, RF, XT).

If these solutions are not enough to solve your concern, please open a separate github issue stating how delayed model saving could assist your current workflow in a way that couldn't also be addressed by increasing disk space.

**Comment by itamar-dw** [comment]
@Innixma 
> 'good_quality_faster_inference_only_refit'

Hi, in the version I use (0.5.2) there is no such a preset. Is there another way to achieve the same functionality?

**Comment by Innixma** [comment]
@itamar-dw ```presets="good_quality"``` if using TabularPredictor.
